nengo_loihi/hardware/nxsdk_objects.py

```diff
@@ -332,12 +332,13 @@ def set_axons(self, board, nxsdk_board, spike_input):
             tcore = d_get(tchip, b"bjJDb3Jlcw==")[tcore_idx]
             spikes = axon.map_spikes(input_idxs)
             for input_idx, spike in zip(input_idxs, spikes):
+                self.axon_map.setdefault(input_idx, [])
+
                 if spike is None:
                     # This should not happen, because input slices happen when
                     # connecting into the spike input. If it does, we just skip it.
                     continue  # pragma: no cover
```

Comment:
Would it be possible to write a test for this?

Reply:
Theoretically yes. I had tried before, though, and couldn't. It's not clear to me what's different about the CIFAR-10 example that it hits this. Using the same input shape and first layer shape should do it. One trick is our current `test_conv_deepnet` uses 7x7 Gabor filters for the first layer, whereas the CIFAR-10 uses 3x3 learned filters. I'm not sure if it would still happen with the 7x7. And if it doesn't then I guess we just use random 3x3 filters and make sure it's close in Nengo and NengoLoihi?

Reply:
I found that this case was actually getting hit already in `test_conv_deepnet`, it just never caused an error because those edge neurons don't spike (because MNIST images are all zero around the edges). I've simplified this code to just have one `continue`, so hopefully it's clearer what's going on.